code/modules/awaymissions/maploader/reader.dm

```diff
@@ -220,9 +220,12 @@ var/global/dmm_suite/preloader/_preloader = new
 	var/atom/instance
 	_preloader.setup(attributes, path)
 
-	var/turf/T = locate(x,y,z)
-	if(T)
-		instance = new path (T)//first preloader pass
+	var/turf/oldTurf = locate(x,y,z)
+	if(ispath(path, /turf)) //Turfs use ChangeTurf
+		if(path != oldTurf.type)
+			instance = oldTurf.ChangeTurf(path)
```

Comment:
@PJB3005 LINDA calculates Adjusted turfs in ChangeTurf, i guess it will be okay in case of perfomance if i'll made lighting processing after the fact.